dev_tools/conftest.py

```diff
@@ -29,24 +29,37 @@
 def pytest_configure(config):
     config.addinivalue_line("markers", "slow: mark tests as slow")
     config.addinivalue_line("markers", "weekly: mark tests as run only by weekly automation")
+    config.addinivalue_line("markers", "rigetti_integration: tests that connect to Quil compiler or QVM.")
 
 
 def pytest_collection_modifyitems(config, items):
     markexpr = config.option.markexpr
     if markexpr:
         return  # let pytest handle this
 
-    skip_slow_marker = pytest.mark.skip(reason='slow marker not selected')
+    # do not skip slow tests if --enable-slow-tests is passed
+    if not config.getoption("--enable-slow-tests"):
+        skip_slow_tests = pytest.mark.skip(reason="slow tests are disabled (use --enable-slow-tests to enable)")
+        for item in items:
+            if "slow" in item.keywords:
+                item.add_marker(skip_slow_tests)
+    # do not skip integration tests if --rigetti-integration option passed
+    if config.getoption('--rigetti-integration'):
+        return
```

Comment:
The early return here will also enable tests with the `weekly` mark, because their deselection happens later on.